### PR TITLE
LASzip{,2}: rename to laszip{,_2}

### PR DESCRIPTION
**Pull request by quantenzitrone**
## Description of changes

renamed to fit package naming conventions
split of https://github.com/NixOS/nixpkgs/pull/319219

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/libLAS/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchurl, fetchpatch, boost, cmake, libgeotiff, libtiff, LASzip2, fixDarwinDylibNames }:
+{ lib, stdenv, fetchurl, fetchpatch, boost, cmake, libgeotiff, libtiff, laszip2, fixDarwinDylibNames }:
```
Comment (eclairevoyant) [inline pkgs/development/libraries/libLAS/default.nix:1]:
```suggestion
{ lib, stdenv, fetchurl, fetchpatch, boost, cmake, libgeotiff, libtiff, laszip_2, fixDarwinDylibNames }:
```

and through the rest of the file

Comment (quantenzitrone) [reply pkgs/development/libraries/libLAS/default.nix]: my bad

**Comment by quantenzitrone** [comment]
it runs into infinite recursion error because the python module is also called laszip
what does one do in this case?

**Comment by wegank** [comment]
Do `inherit (pkgs) cmake ninja laszip;` here.

https://github.com/NixOS/nixpkgs/blob/66540e5f9e118cd0ceb0498cc12e15c3fb0c1ab0/pkgs/top-level/python-packages.nix#L6534-L6536

**Comment by eclairevoyant** [comment]
Wouldn't that break splicing? We should probably pull from `__splicedPackages`

**Comment by quantenzitrone** [comment]
like that?

**Comment by imincik** [comment]
Change looks good to me. I am running nixpkgs-review on x86_64-linux now ...

**Comment by imincik** [comment]
Result of `nixpkgs-review pr 320290` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>cloudcompare</li>
    <li>entwine</li>
    <li>grass</li>
    <li>laszip</li>
    <li>laszip_2</li>
    <li>libLAS</li>
    <li>pdal</li>
    <li>python311Packages.laspy</li>
    <li>python311Packages.laspy.dist</li>
    <li>python311Packages.laszip</li>
    <li>python311Packages.laszip.dist</li>
    <li>python312Packages.laspy</li>
    <li>python312Packages.laspy.dist</li>
    <li>python312Packages.laszip</li>
    <li>python312Packages.laszip.dist</li>
    <li>qgis</li>
    <li>qgis-ltr</li>
  </ul>
</details>

**Review by MatthewCroughan — APPROVED** [review]

**Review by imincik — APPROVED** [review]
Diff looks good and nixpkgs-review built all packages.

**Pull request merged**